### DISCUSSION TITLE
Creating a pane-selection/jump plugin

### DISCUSSION THREAD

**Discussion by meepleek** [zellij-org/zellij#3219] [Q&A]
Wezterm has a nice action to select a pane by overlaying a label over each pane and then just pressing the key for the pane to jump to: https://wezfurlong.org/wezterm/config/lua/keyassignment/PaneSelect.html?h=select+pane .  
I quite miss that and would like to recreate it as a plugin.  

My thinking was the plugin would
- suppress the plugin pane
- create a floating pane for each pane in the current tab using the current panes for positioning the floating label panes
- set text of the label pane just to the label & make the pane not focusable

However, I was not able to figure some stuff out
- is it possible to move & resize panes from plugins?
  - There's the `resize_focused_pane` command. Would I have to listen to the `PaneUpdate` event, and resize the pane there multiple time to get it to the desired tiny size?
- How do I create _display only floating panes_?
  - `open_terminal_floating`? So far I've been testing out `open_command_pane_floating`, but that seems wrong.

Any pointers for how I could achieve this?

**Comment by giyany** [comment]
Hi, my best advice is to join the Zellij Discord/Matrix. It's good for advice on plugin development.

**Reply by imsnif** [reply]
(better Discord, unfortunately not a lot of plugin developers hang around on Matrix)
